### PR TITLE
Add reporting

### PR DESCRIPTION
**Pull request by Zhi-704**
Added SNS functionality where script searches all subscribed topics in AWS and checks for any trending items (threshold > 100 sales in the past 8 hours). In progress #48. Need to implement testings

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,4 +26,5 @@ __pycache__/
 *.tfvars
 
 # IDE (VSCode)
-.vscode/
\ No newline at end of file
+.vscode/!requirements.txt
+!local_requirements.txt
```
Comment (e-lemma) [inline .gitignore:30]: Why is this here?

File: reporting/report.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+from boto3 import client
+
+
+TRENDING_THRESHOLD = 100
+FILTER_TOPICS_BY = "c11-bandcamp-"
+TRENDING_TIMEFRAME = '8 hours'
+
+
+def get_connection() -> Connection:
+    """gets a connection"""
```
Comment (e-lemma) [inline reporting/report.py:16]:
```suggestion
    """Gets a connection"""
```

File: reporting/report.py [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+                 tag_name, trending_items)
+
+    for item in trending_items:
+        item['type'] = item_type
+
+    return trending_items
+
+
+def notification_system() -> None:
+    conn = get_connection()
```
Comment (e-lemma) [inline reporting/report.py:156]: Docstring needed

File: reporting/requirements.txt
Comment (e-lemma) [inline reporting/requirements.txt:1]: Isn't dotenv needed here?

File: schema/insert.sql
Comment (e-lemma) [inline schema/insert.sql:1]: What are these changes for?

File: .gitignore (hunk 1 above)
Comment (Zhi-704) [reply .gitignore]: Oops, must've forgotten it, removed now

File: reporting/report.py (hunk 2 above)
Comment (Zhi-704) [reply reporting/report.py:18]: Grammar police

File: reporting/report.py (hunk 3 above)
Comment (Zhi-704) [reply reporting/report.py]: Added!

File: schema/insert.sql
Comment (e-lemma) [reply schema/insert.sql:1]: I see.

**Comment by Zhi-704** [comment]
Close #48

File: reporting/requirements.txt
Comment (Zhi-704) [reply reporting/requirements.txt:1]: Added

File: reporting/test_report.py
Comment (e-lemma) [inline reporting/test_report.py:1]: What is your test coverage percentage?

Comment (Zhi-704) [reply reporting/test_report.py:1]: 83%

**Review by e-lemma — APPROVED** [review]
Comments addressed, good to merge.
